test/integration/performance_yaml_test.go

```diff
@@ -0,0 +1,165 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/environment"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/exeutils"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/ActiveState/cli/pkg/projectfile"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/yaml.v2"
+)
+
+// Configuration values for the performance tests
+const (
+	DefaultMaxTime  = 1000 * time.Millisecond
+	DefaultSamples  = 10
+	DefaultVariance = 0.2
+	SecretsVariance = 2.4
+	// Add other configuration values on per-test basis if needed
+)
+
+var (
+	rx = regexp.MustCompile(`Profiling: main took .*\((\d+)\)`)
+)
+
+type PerformanceYamlIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *PerformanceYamlIntegrationTestSuite) startSvc(ts *e2e.Session) {
+	// Start svc first, as we don't want to measure svc startup time which would only happen the very first invocation
+	stdout, stderr, err := exeutils.ExecSimple(ts.SvcExe, []string{"start"}, []string{})
+	suite.Require().NoError(err, fmt.Sprintf("Full error:\n%v\nstdout:\n%s\nstderr:\n%s", errs.JoinMessage(err), stdout, stderr))
+}
+
+func (suite *PerformanceYamlIntegrationTestSuite) TestYamlPerformance() {
+	suite.OnlyRunForTags(tagsuite.Performance)
+	baseline := DefaultMaxTime
+	suite.Run("CallScript", func() {
+		avg := suite.testScriptPerformance("call-script", "Hello World", DefaultSamples, DefaultMaxTime)
```

Comment:
We should calculate the baseline with a clean activestate.yaml that has no other scripts or utilized merged yamls, otherwise that would invalidate part of the performance tests (namely the merged yaml one).